wallet/wallet.c

```diff
@@ -3343,7 +3343,35 @@ wallet_payment_list(const tal_t *ctx,
 	size_t i;
 
 	payments = tal_arr(ctx, const struct wallet_payment *, 0);
-	if (payment_hash) {
+        //FIXME: Make the if-else smaller
+	// A possible solution is divided the string in two part (pre-where and post-where) and
+	// use a small if else to set the remain string, with this method we can have small code.
```

Comment:
Or we can do this :-) Might not work if we extend the query args much more to have the full matrix of combinations.

Reply:
This looks like more of a hack solution instead of a real solution to refactoring the code. I will make some experiments with your feedback in the generic query.

Thanks.